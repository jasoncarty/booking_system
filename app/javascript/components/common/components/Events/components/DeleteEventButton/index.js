import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { sendRequest } from './../../../../utils';

const CONFIRM_MESSAGE = 'Are you sure you want to delete this event?';

class DeleteEventButton extends Component {
  static propTypes = {
    eventId: PropTypes.number.isRequired,
    onSuccess: PropTypes.func.isRequired,
    onError: PropTypes.func.isRequired
  }

  handleDeleteEvent = () => {
    const {
      eventId,
      onSuccess,
      onError
    } = this.props;

    sendRequest({
      url: `/admin/events/${eventId}`,
      type: 'delete',
      onSuccess,
      onError
    });
  }

  handleClick = () => {
    const confirm = window.confirm(CONFIRM_MESSAGE);
    if (confirm) {
      this.handleDeleteEvent();
    }
  }

  render() {
    return (
      <div className="delete-link" onClick={this.handleClick}>
        <button type="button">
          <i className='glyphicon glyphicon-trash'></i>
        </button>
      </div>
    );
  }
}

export default DeleteEventButton;
