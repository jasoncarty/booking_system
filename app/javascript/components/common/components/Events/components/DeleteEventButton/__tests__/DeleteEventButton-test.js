import React from 'react';
import jsdom from 'jsdom';
import { create } from 'react-test-renderer';

import DeleteEventButton from './../index';

const { JSDOM } = jsdom;
const dom = new JSDOM('<!doctype html><html><body><div id="root"></div></body></html>');
global.window = dom.window;
global.window.confirm = jest.fn();

const mockSendRequest = jest.fn();
jest.mock('./../../../../../utils/sendRequest', () => () => mockSendRequest);

const defaultProps = {
  onSuccess: () => {},
  onError: () => {},
  eventId: 1
};

describe('<DeleteEventButton />', () => {
  it('matches snapshot with default props', () => {
    const element = create(
      <DeleteEventButton {...defaultProps} />
    );
    expect(element).toMatchSnapshot();
  });

  it('calls sendRequest', () => {
    const element = create(
      <DeleteEventButton {...defaultProps} />
    );
    const instance = element.getInstance();
    const {
      onSuccess,
      onError,
      eventId
    } = defaultProps;

    instance.handleClick();
    setTimeout(() => {
      expect(mockSendRequest).toBeCalledWith({
        url: `/admin/events/${eventId}`,
        type: 'delete',
        onSuccess,
        onError
      });
    }, 0);
  });
});
