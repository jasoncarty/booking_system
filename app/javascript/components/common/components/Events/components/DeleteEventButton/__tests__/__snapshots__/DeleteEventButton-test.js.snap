// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteEventButton /> matches snapshot with default props 1`] = `
<div
  className="delete-link"
  onClick={[Function]}
>
  <button
    type="button"
  >
    <i
      className="glyphicon glyphicon-trash"
    />
  </button>
</div>
`;
